PRINCIPLES.md

```diff
@@ -1,74 +1,22 @@
-
-- ### Desired State
-
-    The aggregate of all configuration data for a system form its _Desired State_ which is defined as data sufficient to recreate the system so that instances of the system are behaviourally indistinguishable.
+The [desired state](./GLOSSARY.md#desired-state) of a GitOps managed system must be:
 
-- ### Drift
+1. **Declarative**
 
-    When a system's _Actual State_ changes for any reason other than its versioned _Desired State_ declarations having changed, we say that the system has drifted from its _Desired State_.
+    A [system](./GLOSSARY.md#software-system) managed by GitOps must have its desired state expressed [declaratively](./GLOSSARY.md#declarative-description).
 
-- ### Reconciliation
+2. **Versioned and Immutable**
 
-    The process of ensuring that the _Actual State_ of a sytem matches its versioned _Desired State_ declarations.
-    Contrary to CIops, any divergence between the two will trigger reconciliation, regardless of where changes occured.
-    Divergence could be due to the actual state unintentionally _Drifting_ from the desired state declarations, or a new desired state declaration version having been changed intentionally.
+    Desired state is [stored](./GLOSSARY.md#state-store) in a way that enforces immutability, versioning and retains a complete version history.
 
-- ### Software System
+3. **Pulled Automatically**
 
-    One or more Runtime environments consisting of resources under management.
-    In each Runtime, management Agents to act on resources according to security policies.
-    One or more software Repositories for storing deployable artifacts that may be loaded into the runtime environments, eg. configuration files, code, binaries and packages.
-    One or more Administrators who are responsible for operating the runtime environments ie. installing, starting, stopping and updating software, code, configuration, etc.
-    A set of policies controlling access and management of repositories, deployments, runtimes.
+    Software agents automatically pull the desired state declarations from the source.
 
-- ### State Store
+4. **Continuously Reconciled**
 
-    A system for storing immutable versions of _Desired State_ declarations.
-    This state store should provide access control and auditing on the changes to the Desired State.
-    Git is the canonical example used as this State Store, and where GitOps derived its name, but but any other system that meets this criteria may be used.
-    In all cases these must be properly configured, and special precautions must be taken to comply with requirements set out in the GitOps Principles.
+    Software agents [continuously](./GLOSSARY.md#continuous) observe actual system state and [attempt to apply](./GLOSSARY.md#reconciliation) the desired state.
```

Comment:
The `attempt to apply` might not communicate the imperative need for the software agent to actually work towards the reconciliation. 

What about:
...observe actual system state and on [drift](./GLOSSARY.md#drift) [reconcile](./GLOSSARY.md#reconciliation) towards the desired state.

Reply:
I'd like to hear what you think about this section of the PR description above: "items left out of this PR"￼ > "existing principal 5"￼ > "5. Operated in a closed loop"? This is the principle that addresses a feedback loop based on configurations in the desired state and the output of attempts to apply to the actual state.￼

Reply:
Sounds like this is not a blocker for RC 1. @williamcaban let's continue to discuss this during the process when we send out RC 1 for feedback for folks. Most likely that will be in the form of a GitHub Discussion ❤️

Reply:
My concern with `attempt to apply` is that it could open the interpretation that a series of remediations were attempted (lets say remediation A & B) and even if not successful it can `attempt` additional ones (eg. remediations C & D) which might lead to a situation of unrecoverable drift state (which brings us back to the snowflakes configurations results and desired state vs running state open to divergence)

How can we phrase it in a way that an `attempt` is not enough, but that it must continue trying remediate?

Reply:
WRT leaving principle 5, agree as it might be covered by principle 3. At least I have not found any extremely compelling example not covers by the other principles. The more concise the better.